src/google/protobuf/descriptor.proto

```diff
@@ -351,6 +351,14 @@ message FileOptions {
   // generated classes from this .proto. There is no default.
   optional string objc_class_prefix = 36;
 
+  // Namespace for generated classes; defaults to the package.
+  optional string csharp_namespace = 37;
+
+  // A nested namespace for the umbrella class.  Helpful for name collisions caused by
+  // umbrella_classname conflicting with an existing type.  This will be automatically
```

Comment:
Quoting what Roger said about umbrella_namespace:
[ROK] it places the SomeFileProto class into a child namespace
like:

namespace SomePackage {
namespace Proto {
class SomFileProto { ... }
}
class SomeMessage { ... }
}
This means that my context-sensitive menu is not cluttered with junk
artifacts from the generation. It also does a better job (IMHO) of
preventing name collisions with messages.